src/feed.rs

```diff
@@ -514,6 +514,28 @@ where
     Ok(extra_nodes)
   }
 
+  /// Audit all data in the feed. Checks that all current data matches
+  /// the hashes in the merkle tree, and clears the bitfield if not.
+  /// The tuple returns is (valid_blocks, invalid_blocks)
+  pub fn audit(&mut self) -> Result<(usize, usize)> {
+    let mut valid_blocks = 0;
+    let mut invalid_blocks = 0;
+    for index in 0..self.length {
+      if self.bitfield.get(index) {
+        let node = self.storage.get_node(2 * index)?;
+        let data = self.storage.get_data(index)?;
+        let data_hash = Hash::from_leaf(&data);
+        if node.hash == data_hash.as_bytes() {
+          valid_blocks += 1;
+        } else {
+          invalid_blocks += 1;
+          self.bitfield.set(index, false);
+        }
+      }
+    }
+    Ok((valid_blocks, invalid_blocks))
```

Comment:
I wonder if we should return a struct here instead. It would allow us to expose methods such as `feed.audit().is_valid()` to check that there are no invalid blocks in the feed at all.

Reply:
I was thinking about that as I designed it; it also ensures you can never accidentally write `let (invalid_blocks, valid_blocks) = feed.audit().unwrap()`. I hadn't seen custom return types elsewhere in the code, so I had avoided it, but I'll add one today after work